src/screens/DiscoveryScreenV2/fragment/TopicFragment.js

```diff
@@ -66,39 +78,81 @@
     parseToken();
   }, []);
 
-  const handleFollow = async (from, willFollow, item, index) => {
+  const exhangeFollower = (newTopicLists, willFollow, topicId) => {
+    const indexTopic = newTopicLists.findIndex((item) => item.topic_id === topicId);
+    newTopicLists[indexTopic].following = !!willFollow;
+    newTopicLists[indexTopic].user_id_follower = myId;
+    return newTopicLists[indexTopic];
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `exhangeFollower` function seems to mutate the `newTopicLists` array directly, which is not recommended in React as it can lead to unexpected behavior. Instead, consider creating a copy of the array before making changes.
```diff
-   const exhangeFollower = (newTopicLists, willFollow, topicId) => {
-     const indexTopic = newTopicLists.findIndex((item) => item.topic_id === topicId);
-     newTopicLists[indexTopic].following = !!willFollow;
-     newTopicLists[indexTopic].user_id_follower = myId;
-     return newTopicLists[indexTopic];
-   };

+   const exhangeFollower = (newTopicLists, willFollow, topicId) => {
+     const updatedTopics = [...newTopicLists];
+     const indexTopic = updatedTopics.findIndex((item) => item.topic_id === topicId);
+     updatedTopics[indexTopic].following = !!willFollow;
+     updatedTopics[indexTopic].user_id_follower = myId;
+     return updatedTopics[indexTopic];
+   };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->